### DISCUSSION TITLE
Develop Application like GEE app

### DISCUSSION THREAD

**Discussion by Jsdogbey** [microsoft/PlanetaryComputer#97] [Q&A]
Hi All,

I am currently new to planetary computer and will be grateful for your help.

Is there a way to develop an application in a planetary computer hub just like Google Earth Engine does so that you can share your result/analysis with the public?

I would be grateful for your advice.

Thanks

Best regards
John Dogbey

**Comment by TomAugspurger** [comment]
It's probably worth thinking through exactly what requirements your "application" needs. Going in roughly increasing order of complexity:

1. Are you just using the Planetary Computer's STAC and Data APIs, accessed using javascript? This is similar to the [Planetary Computer Explorer](https://planetarycomputer.microsoft.com/docs/overview/explorer/) which is a React application deployed on Azure.
2. Are you just using the `data` API, but need to expose some sort of computation environment? Perhaps you could use [Pyodide](https://pyodide.org/en/stable/), which avoids having to set up compute infrastructure in Azure but there are some limitations on what all can run in the browser. https://tomaugspurger.github.io/pc-data-api/lab?path=pc-data-api.ipynb is an example using the Planetary Computer APIs and data from your local browser.
3. There are a bunch of dashboarding frameworks available ([voila](https://voila.readthedocs.io/en/stable/using.html), [panel](https://panel.holoviz.org/), [streamlit](https://streamlit.io/), [Shiny](https://shiny.rstudio.com/)). I don't have a ton of experience with those, so I'm not sure how each is best deployed.
4. Do you need to provide a computation environment and need to control the packages and content, but your workload isn't too computationally demanding? If so, https://mybinder.org/ might be a good option. The compute will be limited and outside of Azure though, which might be a blocker.
5. If your potential users have Planetary Computer accounts, you could [generate a gitpuller link](https://jupyterhub.github.io/nbgitpuller/link) and share that. That's what we do for the ["Launch in Hub"](https://planetarycomputer.microsoft.com/dataset/goes-cmi#Example-Notebook) links from our dataset pages.

So overall there's a lot of options, none of which is as convenient as just sharing a link to an Earth Engine application. Hopefully this is something that'll be easier to do in the future. Let us know if you end up building anything that's shareable!

**Reply by Jsdogbey** [reply]
@TomAugspurger, I will explore your options and share with you the result. Thanks so much.
